gnovm/pkg/gnolang/alloc.go

```diff
@@ -1,14 +1,19 @@
 package gnolang
 
-import "reflect"
+import (
+	"fmt"
+	"reflect"
+)
 
 // Keeps track of in-memory allocations.
 // In the future, allocations within realm boundaries will be
 // (optionally?) condensed (objects to be GC'd will be discarded),
 // but for now, allocations strictly increment across the whole tx.
 type Allocator struct {
-	maxBytes int64
-	bytes    int64
+	m          *Machine
```

Comment:
Instead of m, a callback function to call for garbage collect namely m.GarbageCollect().
Allows the gc to be wrapped and decorated, more loosely coupled.

Reply:
fixed.